src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2819,308 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
+            {
+                if (expectedSuccessKeys.size() > 0)
+                {
+                    REQUIRE(fabs(ltx2.getPrefetchHitRate() - 1.0f) < 0.0001f);
+                }
+                REQUIRE(numLoadedHot == 0);
+            }
+            else
+            {
+                REQUIRE(fabs(ltx2.getPrefetchHitRate() - 1.0f) > 0.0001f);
+                if (expectedSuccessKeys.size() > 0)
+                {
+                    REQUIRE(numLoadedHot >=
+                            expectedSuccessKeys.size() - cacheSize);
+                }
+            }
+
+            // Ensure that the meter is exhausted after loading.
+            // i.e. not all expectedSuccessKeys have remaining capacity.
+            // canLoad should return false for at least some keys in
+            // expectedSuccessKeys now.
+            ensureLedgerKeyMeterCanLoad(lkMeterHot, allEntries,
+                                        /*success keys*/ {},
+                                        /* fail keys */ expectedSuccessKeys);
+        };
+
+        std::vector<LedgerEntry> allEntries;
+        // Overlapping, all keys succeed
+        SECTION("all keys have quota")
+        {
+            // Make sure entries is mix of classic soroban
+            auto tx1Entries =
+                std::vector<LedgerEntry>{entries.at(0), entries.at(1)};
+            auto tx2Entries =
+                std::vector<LedgerEntry>{entries.at(0), entries.at(1)};
+            addTxn(0, tx1Entries);
+            addTxn(-1, tx2Entries);
+            allEntries.insert(allEntries.end(), tx1Entries.begin(),
+                              tx1Entries.end());
+            allEntries.insert(allEntries.end(), tx2Entries.begin(),
+                              tx2Entries.end());
+            checkPrefetch(allEntries);
+        }
+
+        SECTION("dead keys don't affect quota")
+        {
+            auto deadKeys =
+                std::vector<LedgerKey>{deadKeySet.begin(), deadKeySet.end()};
+            auto tx1Entries =
+                std::vector<LedgerEntry>{entries.at(0), entries.at(1)};
+            auto tx2Entries =
+                std::vector<LedgerEntry>{entries.at(0), entries.at(1)};
+            addTxn(0, tx1Entries);
+            addTxn(-1, tx2Entries, deadKeys);
+            allEntries.insert(allEntries.end(), tx1Entries.begin(),
+                              tx1Entries.end());
+            allEntries.insert(allEntries.end(), tx2Entries.begin(),
+                              tx2Entries.end());
+            checkPrefetch(allEntries);
+        }
+
+        SECTION("don't load entries without quota")
+        {
+            auto tx1Entries =
+                std::vector<LedgerEntry>{entries.at(0), entries.at(1)};
+            addTxn(-1, tx1Entries);
+            checkPrefetch(tx1Entries);
+        }
+    };
+
+    SECTION("prefetch with large cache")
+    {
+        // cache size > number of entries
+        runTest(1000);
+    }
+    SECTION("prefetch with small cache")
```

Comment:
I don't think we actually need to check this here. It will simplify the test case, and we already have coverage in the classic prefetch tests. The soroban specific prefetch logic just uses the same cache eviction library as the classic prefetch.